### PR TITLE
Fix new draft bugs

### PR DESCRIPTION
**Pull request by chalo2000**
## Overview

Removed an overlooked line of code from #406 that utterly decimated draft polls. Unsure from @jackthmp's PR what it originally fixed

## Changes Made

- Remove the target for when editing ends on addOptionTextField
- Shorten for loop

## Test Coverage

Manual on iPhone 11 Pro simulator

Please test drafts **on the simulator** and ensure everything works as you would intuitively expect - try to break it

## Related PRs or Issues (delete if not applicable)

#406 

#411 
#414 
#413

### CODE REVIEW

File: Pollo/Views/PollBuilder/MCPollBuilderView.swift [hunk 1]
```diff
@@ -147,9 +147,7 @@ class MCPollBuilderView: UIView {
     }
     
     func updateTotalOptions() {
-        for opt in mcOptionModels {
-            opt.totalOptions = mcOptionModels.count - 1
-        }
+        mcOptionModels.forEach{ $0.totalOptions = mcOptionModels.count - 1 }
```
[the remark this answers is not in the document]
Comment (chalo2000) [reply Pollo/Views/PollBuilder/MCPollBuilderView.swift:150]: bruh

File: Pollo/Views/PollBuilder/MCPollBuilderView.swift [hunk 2]
```diff
@@ -147,9 +147,7 @@ class MCPollBuilderView: UIView {
     }
     
     func updateTotalOptions() {
-        for opt in mcOptionModels {
-            opt.totalOptions = mcOptionModels.count - 1
-        }
+        mcOptionModels.forEach { $0.totalOptions = mcOptionModels.count - 1 }
```
Comment (Omarrasheed) [inline Pollo/Views/PollBuilder/MCPollBuilderView.swift:150]: Love it

**Review by Omarrasheed — APPROVED** [review]
Can u explain what this additional line was doing that broke the code?

**Comment by chalo2000** [comment]
@Omarrasheed 

As @ywsang initially noted, creating a draft poll with 4 MC would crash the app. There were some weird things occurring, like the draft poll would still be created when the app was reopened - it was just an out of bounds error when trying to update a multiple choice's choice after typing into a multiple choice's name (.edittingDidChange) and also the problematic code when the text cursor disappears (.edittingDidEnd). 

I noticed that Yana's bug could only be replicated if you typed in the title first and then made the draft poll instead of creating your choices and then putting a title. This is because if you type in the title first and then create the draft poll when the text cursor is on the last multiple choice, the multiple choices resetting to 2 choices is called first and then the text cursor has to disappear within the bounds of the addOptionTextField, causing a call to update the multiple choice with the old poll's information (and last index). For the same reason, whenever a MC cell is deleted, the text cursor disappears and causes the mcOptionsModel to update at the wrong point.

I'm not sure again why that line of code was there, the only drafts related fix I saw when @jackthmp added this code was #402 and after doing a quick check (and literally just checking again) I didn't see any color changes with this line removed.

**Review by ywsang — APPROVED** [review]
interesting fix!

**Pull request merged**